block-chain/src/main/java/itx/blockchain/api/LedgerBuilder.java

```diff
@@ -0,0 +1,75 @@
+package itx.blockchain.api;
+
+import itx.blockchain.basic.LedgerImpl;
+
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class LedgerBuilder {
+
+    private String id;
+    private HashValidator hashValidator;
+    private Block genesisBlock;
+    private List<Block> blocks;
+    private MessageDigest messageDigest;
+    private List<DataReader<?>> readers;
+    private List<DataWriter<?>> writers;
+
+    public LedgerBuilder() {
+        this.hashValidator = new HashValidator() {};
+        this.blocks = new ArrayList<>();
+        this.readers = new ArrayList<>();
+        this.writers = new ArrayList<>();
+    }
+
+    public LedgerBuilder setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    public LedgerBuilder setHashValidator(HashValidator hashValidator) {
+        this.hashValidator = hashValidator;
+        return this;
+    }
+
+    public LedgerBuilder setGenesisBlock(Block block) {
+        this.genesisBlock = blocks;
+        return this;
+    }
+
+    public LedgerBuilder addBlocks(Collection<Block> blocks) {
+        this.blocks.addAll(blocks);
+        return this;
+    }
+
+    public LedgerBuilder addBlock(Block block) {
+        this.blocks.add(block);
+        return this;
+    }
+
+    public LedgerBuilder setMessageDigest(MessageDigest messageDigest) {
+        this.messageDigest = messageDigest;
+        return this;
+    }
+
+    public LedgerBuilder addDataReader(DataReader<?> dataReader) {
+        this.readers.add(dataReader);
+        return this;
+    }
+
+    public LedgerBuilder addDataWriter(DataWriter<?> dataWriter) {
+        this.writers.add(dataWriter);
+        return this;
+    }
+
+    public Ledger build() throws InvalidDataBlockException {
+        LedgerImpl ledger = new LedgerImpl(id, hashValidator, messageDigest, readers, writers);
+        for (Block b: blocks) {
+            ledger.addDataBlock(b);
+        };
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Genesis block not handled in build method</b></div>
<div id="fix">

The `build()` method doesn't appear to handle the `genesisBlock` field. Consider adding the genesis block first before adding other blocks.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
        LedgerImpl ledger = new LedgerImpl(id, hashValidator, messageDigest, readers, writers);
        if (genesisBlock != null) {
            ledger.addDataBlock(genesisBlock);
        }
        for (Block b: blocks) {
            ledger.addDataBlock(b);
        };
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/4#issuecomment-2557450846>#0dafab</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged